File: flutter_version/lib/src/features/news/domain/entities/article.dart
import 'package:equatable/equatable.dart';

class Article extends Equatable {
  final String? sourceName;
  final String? author;
  final String? title;
  final String? description;
  final String? url;
  final String? urlToImage;
  final DateTime? publishedAt;

  const Article({
    this.sourceName,
    this.author,
    this.title,
    this.description,
    this.url,
    this.urlToImage,
    this.publishedAt,
  });

  @override
  List<Object?> get props => [sourceName, author, title, description, url, urlToImage, publishedAt];
}

File: swiftui_version/News/News/Features/News/Domain/Entities/Article.swift
import Foundation

/// Domain entity representing a news article
struct Article: Identifiable, Hashable, Equatable {
    let id: UUID
    let sourceName: String?
    let author: String?
    let title: String?
    let description: String?
    let url: String?
    let urlToImage: String?
    let publishedAt: Date?
    
    init(
        id: UUID = UUID(),
        sourceName: String? = nil,
        author: String? = nil,
        title: String? = nil,
        description: String? = nil,
        url: String? = nil,
        urlToImage: String? = nil,
        publishedAt: Date? = nil
    ) {
        self.id = id
        self.sourceName = sourceName
        self.author = author
        self.title = title
        self.description = description
        self.url = url
        self.urlToImage = urlToImage
        self.publishedAt = publishedAt
    }
    
    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }
    
    static func == (lhs: Article, rhs: Article) -> Bool {
        lhs.id == rhs.id
    }
}
